pkg/volume/downwardapi/downwardapi.go

```diff
@@ -156,20 +158,20 @@ func (b *downwardAPIVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
-	data, err := b.collectData()
+	data, mode, err := b.collectData()
```

Comment:
same comments here - pass default here, return a list of structs, including mode per-file.  Unless there's something wrong with that, in which case let's dicsuss :)